packages/pds/src/app-view/proxied/index.ts

```diff
@@ -0,0 +1,519 @@
+import { Server } from '../../lexicon'
+import AppContext from '../../context'
+import { AtpAgent } from '@atproto/api'
+import {
+  FeedViewPost,
+  ThreadViewPost,
+  isReasonRepost,
+  isThreadViewPost,
+} from '../../lexicon/types/app/bsky/feed/defs'
+import { dedupeStrs } from '@atproto/common'
+
+export default function (server: Server, ctx: AppContext) {
+  if (!ctx.cfg.bskyAppViewEndpoint) {
+    throw new Error('Could not find bsky appview endpoint')
+  }
+  const agent = new AtpAgent({ service: ctx.cfg.bskyAppViewEndpoint })
+
+  const headers = (did: string) => {
+    return {
+      headers: { authorization: `Bearer ${did}` },
+    }
+  }
+
+  server.app.bsky.actor.getProfile({
+    auth: ctx.accessVerifier,
+    handler: async ({ auth, params }) => {
+      const requester = auth.credentials.did
+      const res = await agent.api.app.bsky.actor.getProfile(
+        params,
+        headers(requester),
+      )
+      const profile = res.data
+      const muted = await ctx.services
+        .account(ctx.db)
+        .getMute(requester, profile.did)
+      profile.viewer ??= {}
+      profile.viewer.muted = muted
+      return {
+        encoding: 'application/json',
+        body: profile,
+      }
+    },
+  })
+
+  server.app.bsky.actor.getProfiles({
+    auth: ctx.accessVerifier,
+    handler: async ({ auth, params }) => {
+      const requester = auth.credentials.did
+      const res = await agent.api.app.bsky.actor.getProfiles(
```

Comment:
What's the story around propagating errors— does that work relatively well?

Reply:
yeah it actually works flawlessly. it's already the correct error class & everything